sotodlib/io/load_smurf.py

```diff
@@ -2125,20 +2133,32 @@ def _get_timestamps(streams, load_type=None, linearize_timestamps=True):
             if None, uses highest precision version possible. integer values
             will use the TimingParadigm class for indexing
         linearize_timestamps : bool
-          if true and using unix timing, linearize the timing based on the 
+          if true and using unix timing, linearize the timing based on the
           frame counter
     """
     if load_type is None:
         # determine the desired loading type. Expand as logic as
         # data fields develop
         if "primary" in streams:
-            if "UnixTime" in streams["primary"]:
+            if np.diff(io_load.hstack_into(None,
+                                streams["primary"]["Counter0"])).mean() != 0:
+                load_type = TimingParadigm.TimingSystem
+            elif "UnixTime" in streams["primary"]:
                 load_type = TimingParadigm.SmurfUnixTime
             else:
                 load_type = TimingParadigm.G3Timestream
         else:
             load_type = TimingParadigm.G3Timestream
 
+    if load_type == TimingParadigm.TimingSystem:
+        s, ns = split_ts_bits(io_load.hstack_into(None,
+                                streams["primary"]["Counter2"]))
+        # Add 20 years in seconds (accounting for leap years) to handle
+        # offset between EPICS time referenced to 1990 relative to UNIX time.
+        counter2 = s + ns*1e-9 + 5*(4*365 - 1)*24*60*60
```

Comment:
could we just hard code this number since it seems like it's not changing? 

You have double checked that the difference between the unix time and the timing system data is small right? (Making sure this isn't off by some weird amount).

Reply:
That offset evaluates to 630288000.  But I would have expected
```date -u -d '1990-01-01' +%s
631152000
```
The equation should have + 1, not - 1, for the leap years.

Reply:
Whoops, totally found this -1 error before when we were trying this out and you corrected me then but I propagated the error through in my haste. Fixed in [0666539](https://github.com/simonsobs/sotodlib/pull/310/commits/06665399a060ae4f1a6bfc97eb3e4739a3a21911) 

For Katie's comment I'm just going to leave in the math so that it's somewhat human readable how I got that number. We should inspect more when the timing system is all setup with the new software but it appears that the two timestamps (EPICS vs UNIX) are the same up to ~0.2s. Here's some plots showing things:

This plot shows the offset between the EPICS + UNIX timestamps: 
![inspect_offset_zoom](https://user-images.githubusercontent.com/25913911/195352830-c02a177f-4402-4d4b-b083-ec1f39353ff1.png)

This plot shows a histogram of the diff of the timestamps for EPICS (excluding the issues around the PPS reset) and UNIX:
![ts_hist](https://user-images.githubusercontent.com/25913911/195352828-1e3d7dbe-0d83-4a00-959f-7834f685fd75.png)

This is the fixed version of the plot I put above but now the timestamps on the 2 axes match.
![fixed_leap_year](https://user-images.githubusercontent.com/25913911/195352833-d8cb3adb-d155-4014-a79f-8ee2564eedce.png)